src/com/google/common/css/MultipleMappingSubstitutionMap.java

```diff
@@ -76,5 +77,5 @@ public static ValueWithMappings createForSingleMapping(String key, String value)
    * associated with this substitution. These mappings are available as
    * {@link ValueWithMappings#mappings}.
    */
-  public ValueWithMappings getValueWithMappings(String key);
+  public ValueWithMappings getValueWithMappings(String key, @Nullable Map<String, String> inputMappings);
```

Comment:
My first idea was to remove duplicate entries, but it can't work : 

If you have an input map like that : 
{"a": "foo", "b": "bar", ...}

And an output map like that : 
{"a": "foobar", "b": "bar", "c": "foo", ...}

If the entry "a" is removed because it is in two maps. So the "foobar" class name is never renamed.

The result mapping must contain the input map but the result must also use the input mapping if a class names is always renamed and the new generated class names must not be in the input mapping. So I think that you can "post-filter" the output easily.